nix-support/tket.nix

```diff
@@ -0,0 +1,44 @@
+self: super:
+let
+  postFixup = import ./includes-fixup.nix;
+  # only import necessary directories so that changes to unrelated
+  # files inside of ../tket won't trigger a rebuild
+  tket-without-tests = super.stdenv.mkDerivation {
+    name = "tket-sources";
+    phases = [ "installPhase" ];
+    installPhase = ''
+      mkdir -p $out;
+      cp -r ${../tket/cmake} $out/cmake;
+      cp -r ${../tket/include} $out/include;
+      cp -r ${../tket/src} $out/src;
+      cp -r ${../tket/CMakeLists.txt} $out/CMakeLists.txt;
+    '';
+  };
+in {
+  tket = super.stdenv.mkDerivation {
+    name = "tket";
+    src = tket-without-tests;
+    nativeBuildInputs = [ super.cmake ];
+    propagatedBuildInputs = super.tklibs
+      ++ [ super.boost super.symengine super.eigen super.nlohmann_json ];
```

Comment:
Do we specify the versions of these libraries anywhere or does nix just pick up the latest available?

Reply:
The versions of the packages are defined by the nixpkgs commit that the flake.lock file is pinned to.

The typical way of bumping the packages when they're available in a newer version of nixpkgs is to run "nix flake update", which updates the lock file.

If a version is needed that isn't in nixpkgs yet, or if the version in nixpkgs is too new, or if something needs to be tweaked, the derivation of the dependency can be defined in a nix file within tket (this is what I did with jsonschema and SymEngine, for example).